src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs

```diff
@@ -223,7 +223,7 @@ private void AnalyzeFormatArgument(OperationAnalysisContext context, IOperation
                 {
                     context.ReportDiagnostic(formatExpression.CreateDiagnostic(CA2253Rule));
                 }
-                else if (char.IsLower(valueName[0]))
+                else if (valueName.Length > 0 && char.IsLower(valueName[0]))
```

Comment:
Is the expected to be never null? Would we do `!string.IsNullOrEmpty` instead?

Reply:
sure we could switch over to that